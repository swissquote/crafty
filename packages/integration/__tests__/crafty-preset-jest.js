const test = require("ava");
const testUtils = require("../utils");

test.serial("Succeeds without transpiling", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-jest/succeeds");

  const result = await testUtils.run(["test"], cwd);

  t.snapshot(result);
});

test.serial("Creates IDE Integration files", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-jest/ide", [
    "jest.config.js",
    ".gitignore"
  ]);

  const result = await testUtils.run(["ide"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);
  t.snapshot(testUtils.readForSnapshot(cwd, "jest.config.js"));

  t.snapshot(testUtils.readForSnapshot(cwd, ".gitignore"));
});

test.serial("Creates IDE Integration files with Babel", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-jest/ide-babel", [
    "jest.config.js",
    ".gitignore"
  ]);

  const result = await testUtils.run(["ide"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);
  t.snapshot(testUtils.readForSnapshot(cwd, "jest.config.js"));
});

test.serial("Succeeds with typescript", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-jest/typescript");

  const result = await testUtils.run(["test"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);
});

test.serial("Succeeds with babel", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-jest/babel");

  const result = await testUtils.run(["test"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);
});

test.serial("Fails with babel", async t => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-jest/babel-fails"
  );

  const result = await testUtils.run(["test"], cwd);

  t.snapshot(result);
  t.is(result.status, 1);
});

test.serial("Succeeds with babel and React", async t => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-jest/babel-react"
  );

  const result = await testUtils.run(["test"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);
});

test.serial("Succeeds with esm module", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-jest/esm");

  const result = await testUtils.run(["test"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);
});

test.serial("Succeeds with esm module and babel", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-jest/esm-babel");

  const result = await testUtils.run(["test"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);
});
