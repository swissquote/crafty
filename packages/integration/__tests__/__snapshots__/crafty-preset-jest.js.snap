// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates IDE Integration files 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
Written jest.config.js
Written .gitignore
",
}
`;

exports[`Creates IDE Integration files 2`] = `
"// This configuration was generated by Crafty
// This file is generated to improve IDE Integration
// You don't need to commit this file, nor need it to run \`crafty test\`

module.exports = {
    \\"moduleDirectories\\": [ /* Ignored paths for diff */ ],
    \\"moduleFileExtensions\\": [
        \\"js\\",
        \\"json\\"
    ],
    \\"testPathIgnorePatterns\\": [
        \\"/node_modules/\\",
        \\"__PATH__/packages/integration/fixtures/crafty-preset-jest/ide/dist\\"
    ],
    \\"moduleNameMapper\\": {
        \\"\\\\\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$\\": \\"__PATH__/packages/crafty-preset-jest/src/file-mock.js\\",
        \\"\\\\\\\\.(css|less|sass|scss)$\\": \\"__PATH__/packages/crafty-preset-jest/src/style-mock.js\\"
    },
    \\"bail\\": true,
    \\"roots\\": [
        \\"__PATH__/packages/integration/fixtures/crafty-preset-jest/ide\\"
    ],
    \\"transform\\": {
        \\"[/\\\\\\\\\\\\\\\\]node_modules[/\\\\\\\\\\\\\\\\].+\\\\\\\\.m?js$\\": \\"__PATH__/packages/crafty-preset-jest/src/esm-transformer.js\\"
    },
    \\"globals\\": {},
    \\"transformIgnorePatterns\\": [],
    \\"testRegex\\": \\"(/__tests__/.*|(\\\\\\\\.|/)(test|spec))\\\\\\\\.(js)$\\"
};
"
`;

exports[`Creates IDE Integration files 3`] = `
"/jest.config.js
"
`;

exports[`Creates IDE Integration files with Babel 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
Written jest.config.js
Written .gitignore
",
}
`;

exports[`Creates IDE Integration files with Babel 2`] = `
"// This configuration was generated by Crafty
// This file is generated to improve IDE Integration
// You don't need to commit this file, nor need it to run \`crafty test\`

module.exports = {
    \\"moduleDirectories\\": [ /* Ignored paths for diff */ ],
    \\"moduleFileExtensions\\": [
        \\"js\\",
        \\"json\\"
    ],
    \\"testPathIgnorePatterns\\": [
        \\"/node_modules/\\",
        \\"__PATH__/packages/integration/fixtures/crafty-preset-jest/ide-babel/dist\\"
    ],
    \\"moduleNameMapper\\": {
        \\"\\\\\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$\\": \\"__PATH__/packages/crafty-preset-jest/src/file-mock.js\\",
        \\"\\\\\\\\.(css|less|sass|scss)$\\": \\"__PATH__/packages/crafty-preset-jest/src/style-mock.js\\"
    },
    \\"bail\\": true,
    \\"roots\\": [
        \\"__PATH__/packages/integration/fixtures/crafty-preset-jest/ide-babel\\"
    ],
    \\"transform\\": {
        \\"[/\\\\\\\\\\\\\\\\]node_modules[/\\\\\\\\\\\\\\\\].+\\\\\\\\.m?js$\\": \\"__PATH__/packages/crafty-preset-jest/src/esm-transformer.js\\"
    },
    \\"globals\\": {},
    \\"transformIgnorePatterns\\": [],
    \\"testRegex\\": \\"(/__tests__/.*|(\\\\\\\\.|/)(test|spec))\\\\\\\\.(js)$\\"
};
"
`;

exports[`Fails with babel 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
FAIL src/__tests__/math.js
  ✕ adds two numbers (__ms)

  ● adds two numbers

    expect(received).toEqual(expected) // deep equality

    Expected: 5
    Received: 4

      3 |
      4 | it(\\"adds two numbers\\", () => {
    > 5 |     expect(add(2,2)).toEqual(5);
        |                      ^
      6 | })

      ...stacktrace...

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
}
`;

exports[`Succeeds with babel 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
PASS src/__tests__/math.js
  ✓ adds two numbers (__ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
}
`;

exports[`Succeeds with babel and React 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
PASS src/__tests__/MyComponent.jsx
  <MyComponent />
    ✓ renders three <Foo /> components (__ms)
    ✓ renders an \`.icon-star\` (__ms)
    ✓ renders children when passed in (__ms)
  <Counter />
    ✓ can render and update a counter (__ms)

Test Suites: 1 passed, 1 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
}
`;

exports[`Succeeds with esm module 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
PASS src/__tests__/math.js
  ✓ adds two numbers (__ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
}
`;

exports[`Succeeds with esm module and babel 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
PASS src/__tests__/math.js
  ✓ adds two numbers (__ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
}
`;

exports[`Succeeds with typescript 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
PASS src/__tests__/math.ts
  ✓ adds two numbers (__ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
}
`;

exports[`Succeeds without transpiling 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
PASS src/__tests__/math.js
  ✓ adds two numbers (__ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        _____s
Ran all test suites.
",
}
`;
